### PR TITLE
Cryptography

### LINKED ISSUE #937: Cryptography
# Aim

 The objective of this project is to encode and decode messages using a common key.

# Details

 In this project, 
-Users have to enter the message to encode or decode. 
-Users have to select the mode to choose the encoding and decoding process. 
-The same key will be used to process the encoding and decoding for the same message.

### PR DESCRIPTION
**Pull request by codebuzzer01**
# Description
This objective of this project is to encode and decode messages using a common key. This project is built using the Tkinter and base64 library.

## Fixes #937

Replace `issue_no` with the issue number which is fixed in this PR

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [X] Yes
- [ ] No

## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update
- [X] Documentation Update

## Checklist:

- [X] My code follows the style guidelines(Clean Code) of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have created a helpful and easy to understand `README.md`
- [X] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [X] My changes generate no new warnings
- [X] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.
- Screenshot (195)
Screenshot (196)

### CODE REVIEW

**Comment by codebuzzer01** [comment]
Hey @kaustubhgupta  pls review it.

File: Cryptography/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+### This objective of this project is to encode and decode messages using a common key. This project will be built using the Tkinter and base64 library.
```
Comment (kaustubhgupta) [inline Cryptography/README.md:1]: The readme file should have at least 2 -3 headings to describe different sections. My suggested sections: Title of the script, small description, any requirements, how to run, and at the last author name with your GitHub as a hyperlink (you should take credit for your work!)

You can check out other merged scripts readme for implementation or use this readme template as it is:
https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Code works fine but changes in readme:

File: Cryptography/README.md (hunk 1 above)
Comment (codebuzzer01) [reply Cryptography/README.md:1]: ok @kaustubhgupta  I will improve that .

**Comment by codebuzzer01** [comment]
Hey @kaustubhgupta  one check failed. How should I resolve it?

File: Cryptography/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+### TITLE- CRYPTOGRAPHY
```
Comment (kaustubhgupta) [inline Cryptography/README.md:1]:
```suggestion
# CRYPTOGRAPHY

```

File: Cryptography/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+### TITLE- CRYPTOGRAPHY
+
+### DESCRIPTION- This objective of this project is to encode and decode messages using a common key. In this project, users have to enter the message to encode or decode. Users have to select the mode to choose the encoding and decoding process. The same key must be used to process the encoding and decoding for the same message.
```
Comment (kaustubhgupta) [inline Cryptography/README.md:3]:
```suggestion
## DESCRIPTION
The objective of this project is to encode and decode messages using a common key. In this project, users have to enter the message to encode or decode. Users have to select the mode to choose the encoding and decoding process. The same key must be used to process the encoding and decoding for the same message.

```

File: Cryptography/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+### TITLE- CRYPTOGRAPHY
+
+### DESCRIPTION- This objective of this project is to encode and decode messages using a common key. In this project, users have to enter the message to encode or decode. Users have to select the mode to choose the encoding and decoding process. The same key must be used to process the encoding and decoding for the same message.
+
+### PROJECT REQUISITES- To build this project we will use basic concept of python, Tkinter, and base64 library.
```
Comment (kaustubhgupta) [inline Cryptography/README.md:5]:
```suggestion
## PROJECT REQUISITES
To build this project we will use the basic concept of python, Tkinter, and base64 library.

```

File: Cryptography/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+### TITLE- CRYPTOGRAPHY
+
+### DESCRIPTION- This objective of this project is to encode and decode messages using a common key. In this project, users have to enter the message to encode or decode. Users have to select the mode to choose the encoding and decoding process. The same key must be used to process the encoding and decoding for the same message.
+
+### PROJECT REQUISITES- To build this project we will use basic concept of python, Tkinter, and base64 library.
+### -Tkinter is a standard GUI python library.
```
Comment (kaustubhgupta) [inline Cryptography/README.md:6]:
```suggestion
- Tkinter is a standard GUI python library.

```

File: Cryptography/README.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+### TITLE- CRYPTOGRAPHY
+
+### DESCRIPTION- This objective of this project is to encode and decode messages using a common key. In this project, users have to enter the message to encode or decode. Users have to select the mode to choose the encoding and decoding process. The same key must be used to process the encoding and decoding for the same message.
+
+### PROJECT REQUISITES- To build this project we will use basic concept of python, Tkinter, and base64 library.
+### -Tkinter is a standard GUI python library.
+### -base64 module provides a function to encode the binary data to ASCII characters and decode that ASCII characters back to binary data.
```
Comment (kaustubhgupta) [inline Cryptography/README.md:7]:
```suggestion
- base64 module provides a function to encode the binary data to ASCII characters and decode that ASCII characters back to binary data.

```

File: Cryptography/README.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+### TITLE- CRYPTOGRAPHY
+
+### DESCRIPTION- This objective of this project is to encode and decode messages using a common key. In this project, users have to enter the message to encode or decode. Users have to select the mode to choose the encoding and decoding process. The same key must be used to process the encoding and decoding for the same message.
+
+### PROJECT REQUISITES- To build this project we will use basic concept of python, Tkinter, and base64 library.
+### -Tkinter is a standard GUI python library.
+### -base64 module provides a function to encode the binary data to ASCII characters and decode that ASCII characters back to binary data.
+### -Tkinter is a standard GUI python library base64 module provides a function to encode the binary data to ASCII characters and decode that ASCII characters back to binary data.
```
Comment (kaustubhgupta) [inline Cryptography/README.md:8]:
```suggestion
- Tkinter is a standard GUI python library base64 module that provides a function to encode the binary data to ASCII characters and decode that ASCII characters back to binary data.

```

File: Cryptography/README.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
+### TITLE- CRYPTOGRAPHY
+
+### DESCRIPTION- This objective of this project is to encode and decode messages using a common key. In this project, users have to enter the message to encode or decode. Users have to select the mode to choose the encoding and decoding process. The same key must be used to process the encoding and decoding for the same message.
+
+### PROJECT REQUISITES- To build this project we will use basic concept of python, Tkinter, and base64 library.
+### -Tkinter is a standard GUI python library.
+### -base64 module provides a function to encode the binary data to ASCII characters and decode that ASCII characters back to binary data.
+### -Tkinter is a standard GUI python library base64 module provides a function to encode the binary data to ASCII characters and decode that ASCII characters back to binary data.
+
+### PROJECT STRUCTURE- These are the step to build message encode – decode python project-
```
Comment (kaustubhgupta) [inline Cryptography/README.md:10]:
```suggestion
## PROJECT STRUCTURE
These are the step to build message encode – decode python project-

```

File: Cryptography/README.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+### DESCRIPTION- This objective of this project is to encode and decode messages using a common key. In this project, users have to enter the message to encode or decode. Users have to select the mode to choose the encoding and decoding process. The same key must be used to process the encoding and decoding for the same message.
+
+### PROJECT REQUISITES- To build this project we will use basic concept of python, Tkinter, and base64 library.
+### -Tkinter is a standard GUI python library.
+### -base64 module provides a function to encode the binary data to ASCII characters and decode that ASCII characters back to binary data.
+### -Tkinter is a standard GUI python library base64 module provides a function to encode the binary data to ASCII characters and decode that ASCII characters back to binary data.
+
+### PROJECT STRUCTURE- These are the step to build message encode – decode python project-
+### -Import module
```
Comment (kaustubhgupta) [inline Cryptography/README.md:11]:
```suggestion
- Import module

```

File: Cryptography/README.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+### -Create display window
```
Comment (kaustubhgupta) [inline Cryptography/README.md:12]:
```suggestion
- Create display window

```

File: Cryptography/README.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+### -Define function
```
Comment (kaustubhgupta) [inline Cryptography/README.md:13]:
```suggestion
- Define function

```

File: Cryptography/README.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+### -Define labels and buttons
```
Comment (kaustubhgupta) [inline Cryptography/README.md:14]:
```suggestion
- Define labels and buttons

```

File: Cryptography/README.md [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+
+
+
+### AUTHOR NAME- ANUSHKA CHITRANSHI
```
Comment (kaustubhgupta) [inline Cryptography/README.md:18]:
```suggestion
## AUTHOR NAME
[ANUSHKA CHITRANSHI](https://github.com/codebuzzer01)

```

File: Cryptography/README.md [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+### GITHUB PROFILE- https://github.com/codebuzzer01
```
Comment (kaustubhgupta) [inline Cryptography/README.md:19]: Remove this line

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
For now, look at your logs from [here](https://deepsource.io/gh/avinashkranjan/Amazing-Python-Scripts/run/5a446d5d-d86e-41d0-81f1-7c0f1904d429/python/) and try to make the changes suggested. Also a lot readme changes:

**Comment by codebuzzer01** [comment]
Hey @kaustubhgupta  now all checks are passed and have updated README file as well. So pls review it.

File: Cryptography/README.md [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+
+- Import module
+
+
+- Create display window
+
+- Define function
+
+- Define labels and buttons
+
```
Comment (kaustubhgupta) [inline Cryptography/README.md:29]: remove these extra blank lines, only 2 blank lines are enough

File: Cryptography/README.md [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+- base64 module provides a function to encode the binary data to ASCII characters and decode that ASCII characters back to binary data.
+
+- Tkinter is a standard GUI python library base64 module that provides a function to encode the binary data to ASCII characters and decode that ASCII characters back to binary data.
+
+
+## PROJECT STRUCTURE
+These are the step to build message encode – decode python project-
+
+- Import module
+
```
Comment (kaustubhgupta) [inline Cryptography/README.md:22]: only one blank line required here

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by codebuzzer01** [comment]
@kaustubhgupta  review it!

**Review by kaustubhgupta — APPROVED** [review]

**Comment by codebuzzer01** [comment]
@antrikshmisri  and @santushtisharma10  Pls review it.

**Review by santushtisharma10 — APPROVED** [review]

**Review by antrikshmisri — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM

**Pull request merged**